src/Hyperbee.Json/Nodes/JsonPath.cs

```diff
@@ -30,34 +30,20 @@
 //
 #endregion
 
-using System.Collections.Immutable;
-using System.Globalization;
 using System.Text.Json.Nodes;
-using System.Text.RegularExpressions;
 using Hyperbee.Json.Evaluators;
-using Hyperbee.Json.Extensions;
-using Hyperbee.Json.Memory;
 using Hyperbee.Json.Tokenizer;
 
 namespace Hyperbee.Json.Nodes;
 // https://ietf-wg-jsonpath.github.io/draft-ietf-jsonpath-base/draft-ietf-jsonpath-base.html
 // https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base
 
-public sealed partial class JsonPathNode
+public sealed class JsonPathNode
 {
     public static IJsonPathScriptEvaluator<JsonNode> DefaultEvaluator { get; set; } = new JsonPathCSharpNodeEvaluator();
     private readonly IJsonPathScriptEvaluator<JsonNode> _evaluator;
 
-    // generated regex
-
-    [GeneratedRegex( "^(-?[0-9]*):?(-?[0-9]*):?(-?[0-9]*)$" )]
-    private static partial Regex RegexSlice();
-
-    [GeneratedRegex( @"^\?\((.*?)\)$" )]
-    private static partial Regex RegexPathFilter();
-
-    [GeneratedRegex( @"^[0-9*]+$" )]
-    private static partial Regex RegexNumber();
+    private readonly JsonNodePathVisitor _visitor = new();
```

Comment:
I moved all the JSONNode-specific code into a visitor that only gets newed up once.